### PR TITLE
Fix discriminator mapping usages in ThreeDsServers [Part 1/3]

### PR DESCRIPTION
**Pull request by jvc9109**
## Summary
- Simplification of discriminators with only 1 option.
- Reduction of indirection levels

Please checkout the style inconsistency between 3DS config style. For example A1Gateway vs eMerchantPay

https://preview.redoc.ly/all-rebilly/fix-discriminator-mappings/tag/Gateway-accounts/operation/PostGatewayAccount

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/BankAccounts/AccountNumberTypes/BBANType.yaml [hunk 1]
```diff
@@ -8,55 +8,51 @@ required:
   - routingNumber
   - accountType
   - billingAddress
-allOf:
```
Comment (ashkarpetin) [inline openapi/components/schemas/BankAccounts/AccountNumberTypes/BBANType.yaml:11]: Hi this is unrelated to threedsecure

Comment (jvc9109) [reply openapi/components/schemas/BankAccounts/AccountNumberTypes/BBANType.yaml:11]: Ops, you are right

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/GatewayAccountConfig/Clearhaus.yaml [hunk 2]
```diff
@@ -16,4 +16,25 @@ allOf:
...
+            type: string
+            description: Merchant plug-in name.
+            enum:
+              - Clearhaus3dsServer
+          merchantId:
+            type: string
+            description: Merchant account ID or Card Acceptor ID provided by the acquiring bank.
+          acquirerBin:
+            type: string
+            description: Acquiring institution identification code.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/Clearhaus.yaml:36]: suggestion: suggestion for readability

(Series of nouns can be difficult to read)

```suggestion
            description: Identification code of the acquiring institution.
```

File: openapi/components/schemas/GatewayAccountConfig/DataCash.yaml [hunk 3]
```diff
@@ -62,4 +62,13 @@ allOf:
             minimum: 0
             default: 0
       threeDSecureServer:
-        $ref: ../ThreeDSecureServers/DataCash3dsServers/DataCash3dsServers.yaml
+        type: object
+        description: DataCash 3DS servers.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/DataCash.yaml:66]: question: should this be singular or plural 
```suggestion
        description: DataCash 3DS server.
```

File: openapi/components/schemas/GatewayAccountConfig/DataCash.yaml [hunk 4]
```diff
@@ -62,4 +62,13 @@ allOf:
...
+        required:
+          - name
+        properties:
+          name:
+            description: Merchant plug-in name.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/DataCash.yaml:71]: suggestion: suggestion for readability (series of nouns)
```suggestion
            description: Name of the merchant plug-in.
```

File: openapi/components/schemas/GatewayAccountConfig/Ilixium.yaml [hunk 5]
```diff
@@ -45,4 +45,11 @@ allOf:
               - itix
               - tpg
       threeDSecureServer:
-        $ref: ../ThreeDSecureServers/Ilixium3dsServers/Ilixium3dsServers.yaml
+        type: object
+        description: Ilixium 3DS servers.
+        properties:
+          name:
+            description: Merchant plug-in name.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/Ilixium.yaml:52]: suggestion: suggestion for readability (series of nouns)
```suggestion
            description: Name of the merchant plug-in.
```

File: openapi/components/schemas/GatewayAccountConfig/Ilixium.yaml [hunk 6]
```diff
@@ -45,4 +45,11 @@ allOf:
               - itix
               - tpg
       threeDSecureServer:
-        $ref: ../ThreeDSecureServers/Ilixium3dsServers/Ilixium3dsServers.yaml
+        type: object
+        description: Ilixium 3DS servers.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/Ilixium.yaml:49]: question: should this be singular or plural
```suggestion
        description: Ilixium 3DS server.
```

File: openapi/components/schemas/ThreeDSecureServers/EMerchantPay3dsServers/eMerchantPay3dsServer.yaml [hunk 7]
```diff
@@ -1,4 +1,10 @@
...
 description: eMerchantPay 3DS server.
-allOf:
-  - $ref: ./eMerchantPay3dsServers.yaml
-
+required:
+  - name
+properties:
+  name:
+    type: string
+    description: Merchant plug-in name.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/EMerchantPay3dsServers/eMerchantPay3dsServer.yaml:8]: suggestion: suggestion for readability (series of nouns)
```suggestion
    description: Name of the merchant plug-in.
```

File: openapi/components/schemas/ThreeDSecureServers/EMerchantPay3dsServers/eMerchantPay3dsServers.yaml [hunk 8]
```diff
@@ -1,16 +1,12 @@
 description: eMerchantPay 3DS servers.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/EMerchantPay3dsServers/eMerchantPay3dsServers.yaml:1]: question: should this be singular or plural? 
```suggestion
description: eMerchantPay 3DS server.
```

File: openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml [hunk 9]
```diff
@@ -1,123 +1,128 @@
...
+    type: string
+    description: Discover/Diners acquirer merchant ID (MID).
+    maxLength: 35
+  acquirerMerchantIdJcb:
+    type: string
+    description: JCB acquirer merchant ID (MID).
+    maxLength: 35
+  merchantName:
+    type: string
+    description: Merchant name.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml:41]: suggestion: suggestion for readability
```suggestion
    description: Name of the merchant.
```

File: openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml [hunk 10]
```diff
@@ -1,123 +1,128 @@
...
+    minLength: 6
+    maxLength: 11
+  merchantAcquirerBinDiscover:
+    type: string
+    description: Discover/Diners acquirer BIN.
+    minLength: 6
+    maxLength: 11
+  merchantAcquirerBinJcb:
+    type: string
+    description: JCB acquirer BIN.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml:65]: suggestion: suggestion for readability (series of nouns)
```suggestion
    description: BIN of the Visa acquirer.
    minLength: 6
    maxLength: 11
  merchantAcquirerBinMastercard:
    type: string
    description: BIN of the Mastercard acquirer.
    minLength: 6
    maxLength: 11
  merchantAcquirerBinAmex:
    type: string
    description: BIN of the American Express acquirer.
    minLength: 6
    maxLength: 11
  merchantAcquirerBinDiscover:
    type: string
    description: BIN of the Discover/Diners acquirer.
    minLength: 6
    maxLength: 11
  merchantAcquirerBinJcb:
    type: string
    description: BIN of the JCB acquirer.
```

File: openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml [hunk 11]
```diff
@@ -1,123 +1,128 @@
...
+    minLength: 6
+    maxLength: 11
+  merchantCountry:
+    type: string
+    description: Merchant country ISO Alpha-2 code format.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml:70]: nitpick: typo
```suggestion
    description: Merchant country in ISO Alpha-2 code format.
```

File: openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml [hunk 12]
```diff
@@ -1,123 +1,128 @@
...
+    maxLength: 2
+    example: US
+  merchantUrl:
+    type: string
+    description: Merchant URL.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml:75]: suggestion: suggestion to provide more detail
```suggestion
    description: URL of the merchant's website.
```

File: openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml [hunk 13]
```diff
@@ -1,123 +1,128 @@
...
+      10: Account funding.
+      11: Quasi-Cash transaction. Indicates the type of 3RI request.
+      28: Prepaid activation and load.
+  declineNotEnrolled:
+    type: boolean
+    description: Specifies whether to decline transactions if a payment card is not enrolled.
+    default: false
+  use3dsForMerchantInitiated:
+    type: boolean
+    description: Use 3DS for merchant initiated transactions.
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml:98]: nitpick: I think this description is missing the `specifies...` part
```suggestion
    description: Specifies whether to use 3DS for merchant initiated transactions.
```

File: openapi/components/schemas/GatewayAccountConfig/Clearhaus.yaml [hunk 14]
```diff
@@ -16,4 +16,25 @@ allOf:
...
+          - apiKey
+        properties:
+          name:
+            type: string
+            description: Merchant plug-in name.
+            enum:
+              - Clearhaus3dsServer
+          merchantId:
+            type: string
+            description: Merchant account ID or Card Acceptor ID provided by the acquiring bank.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/Clearhaus.yaml:33]: nitpick: capitalization 
```suggestion
            description: Merchant account ID or card acceptor ID provided by the acquiring bank.
```

File: openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml [hunk 15]
```diff
@@ -1,123 +1,128 @@
...
+    type: string
+    description: American Express acquirer merchant ID (MID).
+    maxLength: 35
+  acquirerMerchantIdDiscover:
+    type: string
+    description: Discover/Diners acquirer merchant ID (MID).
+    maxLength: 35
+  acquirerMerchantIdJcb:
+    type: string
+    description: JCB acquirer merchant ID (MID).
```
Comment (noon-dawg) [inline openapi/components/schemas/ThreeDSecureServers/ThreeDSecureIO3dsServer.yaml:37]: suggestion: suggestion for readability (series of nouns)
```suggestion
    description: Merchant ID (MID) of the Visa acquirer.
    maxLength: 35
  acquirerMerchantIdMastercard:
    type: string
    description: Merchant ID (MID) of the Mastercard acquirer.
    maxLength: 35
  acquirerMerchantIdAmex:
    type: string
    description: Merchant ID (MID) of the American Express acquirer.
    maxLength: 35
  acquirerMerchantIdDiscover:
    type: string
    description: Merchant ID (MID) of the Discover/Diners acquirer .
    maxLength: 35
  acquirerMerchantIdJcb:
    type: string
    description: Merchant ID (MID) of the JCB acquirer.
```

File: openapi/components/schemas/GatewayAccountConfig/Clearhaus.yaml [hunk 16]
```diff
@@ -16,4 +16,25 @@ allOf:
...
+          acquirerBin:
+            type: string
+            description: Acquiring institution identification code.
+          apiKey:
+            type: string
+            format: password
+            description: API key for 3D Secure server.
```
Comment (noon-dawg) [inline openapi/components/schemas/GatewayAccountConfig/Clearhaus.yaml:40]: nitpick: typo
```suggestion
            description: API key of the 3D Secure server.
```

**Review by noon-dawg — CHANGES_REQUESTED** [review]
Minor nitpicks and suggestions. I think these may have been in the docs already, again I'm taking the opportunity 🙏

**Review by noon-dawg — APPROVED** [review]

**Comment by jvc9109** [comment]
@Rebilly/js @Rebilly/php  please review

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**